### PR TITLE
py-reportlab: add 4.0.4

### PR DESCRIPTION
**Pull request by manuelakuhn**
https://pypi.org/project/reportlab/#files

### CODE REVIEW

**Review by tldahlgren — COMMENTED** [review]
Confirmed the new version sha256.

**Comment by tldahlgren** [comment]
Deferring merge to @adamjstewart and or @pradyunsg

File: var/spack/repos/builtin/packages/py-reportlab/package.py [hunk 1]
```diff
@@ -13,16 +13,19 @@ class PyReportlab(PythonPackage):
...
 
     depends_on("python@3.7:3", when="@3.6.9:", type=("build", "run"))
+    # version restictions were taken over from release 3.4.0 setup.py
     depends_on("py-setuptools@2.2:", type="build")
     depends_on("py-pip@1.4.1:", type="build")
     depends_on("pil@9:", when="@3.6.10:", type=("build", "run"))
     depends_on("pil@2.4.0:", type=("build", "run"))
     depends_on("freetype")
 
+    @when("@:3")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-reportlab/package.py:28]: Is this not needed anymore or does it not work anymore?

Comment (manuelakuhn) [reply var/spack/repos/builtin/packages/py-reportlab/package.py:28]: The `[FREETYPE_PATHS]` option does not exist in `setup.cfg` anymore so the patch does nothing.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**